### ISSUE TITLE
cisco.ios.ios_facts and gather_network_resources: acls 

### ISSUE THREAD

**Issue by joefodor** [ansible-collections/cisco.ios#155]
##### SUMMARY

The gather_network_resourses: acls option on devices with large ACLs errors out with "line 140, in populate_port_protocol\nRuntimeError: maximum recursion depth exceeded in cmp"
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

cisco.ios.ios_facts:
        gather_subset: all
        gather_network_resources: acl
##### ANSIBLE VERSION

```paste below
ansible 2.10.1
  config file = /Users/fodorj/Documents/git/ansible-network_remote_site/ansible.cfg
  configured module search path = ['/Users/fodorj/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/fodorj/Downloads/exit/envs/ansible210/lib/python3.8/site-packages/ansible
  executable location = /Users/fodorj/Downloads/exit/envs/ansible210/bin/ansible
  python version = 3.8.5 (default, Sep  4 2020, 02:22:02) [Clang 10.0.0 ]
```

##### CONFIGURATION

```paste below
DEFAULT_CALLBACK_WHITELIST(/Users/fodorj/Documents/git/ansible-network_remote_site/ansible.cfg) = ['profile_tasks']
HOST_KEY_CHECKING(/Users/fodorj/Documents/git/ansible-network_remote_site/ansible.cfg) = False
PERSISTENT_COMMAND_TIMEOUT(/Users/fodorj/Documents/git/ansible-network_remote_site/ansible.cfg) = 50
PERSISTENT_CONNECT_TIMEOUT(/Users/fodorj/Documents/git/ansible-network_remote_site/ansible.cfg) = 60
```

##### OS / ENVIRONMENT

Cisco 3650 16.9.5

##### STEPS TO REPRODUCE

run the example playbook on a switch with large ACL

```yaml
      cisco.ios.ios_facts:
        gather_subset: all
        gather_network_resources: acls
```

##### EXPECTED RESULTS

"ansible_network_resources": {
            "acls": [
                {
                    "acls": [
                        {
                            "aces": [
                                {
                                    "grant": "permit",
                                    "sequence": 10,
                                    "source": {
                                        "address": "x.x.x.x"
                                    }
                                }
                            ],
                            "acl_type": "standard",
                            "name": "1"

##### ACTUAL RESULTS

```paste below
(ansible210) ➜  ansible-network_remote_site git:(welcome-to-the-suck) ✗ ansible-playbook the_suck.yml -i inventories/prd/hosts --ask-vault-pass -l "sw01-kansascity.wwt.com" -vvv
zx^C [ERROR]: User interrupted execution
(ansible210) ➜  ansible-network_remote_site git:(welcome-to-the-suck) ✗ ansible-playbook the_suck.yml -i inventories/prd/hosts --ask-vault-pass -l "sw01-kansascity.wwt.com" -vvvv
ansible-playbook 2.10.1
  config file = /Users/fodorj/Documents/git/ansible-network_remote_site/ansible.cfg
  configured module search path = ['/Users/fodorj/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/fodorj/Downloads/exit/envs/ansible210/lib/python3.8/site-packages/ansible
  executable location = /Users/fodorj/Downloads/exit/envs/ansible210/bin/ansible-playbook
  python version = 3.8.5 (default, Sep  4 2020, 02:22:02) [Clang 10.0.0 ]
Using /Users/fodorj/Documents/git/ansible-network_remote_site/ansible.cfg as config file
Vault password: 
setting up inventory plugins
host_list declined parsing /Users/fodorj/Documents/git/ansible-network_remote_site/inventories/prd/hosts as it did not pass its verify_file() method
script declined parsing /Users/fodorj/Documents/git/ansible-network_remote_site/inventories/prd/hosts as it did not pass its verify_file() method
auto declined parsing /Users/fodorj/Documents/git/ansible-network_remote_site/inventories/prd/hosts as it did not pass its verify_file() method
Parsed /Users/fodorj/Documents/git/ansible-network_remote_site/inventories/prd/hosts inventory source with ini plugin
Loading collection cisco.ios from /Users/fodorj/.ansible/collections/ansible_collections/cisco/ios
redirecting (type: action) cisco.ios.ios_facts to cisco.ios.ios
Loading callback plugin default of type stdout, v2.0 from /Users/fodorj/Downloads/exit/envs/ansible210/lib/python3.8/site-packages/ansible/plugins/callback/default.py
redirecting (type: callback) ansible.builtin.profile_tasks to ansible.posix.profile_tasks
Loading collection ansible.posix from /Users/fodorj/Downloads/exit/envs/ansible210/lib/python3.8/site-packages/ansible_collections/ansible/posix
Loading callback plugin ansible.posix.profile_tasks of type aggregate, v2.0 from /Users/fodorj/Downloads/exit/envs/ansible210/lib/python3.8/site-packages/ansible_collections/ansible/posix/plugins/callback/profile_tasks.py

PLAYBOOK: the_suck.yml *******************************************************************************************************************************
Positional arguments: the_suck.yml
verbosity: 4
connection: smart
timeout: 10
become_method: sudo
tags: ('all',)
inventory: ('/Users/fodorj/Documents/git/ansible-network_remote_site/inventories/prd/hosts',)
subset: sw01-kansascity.wwt.com
ask_vault_pass: True
forks: 5
1 plays in the_suck.yml

PLAY [getstuff] **************************************************************************************************************************************
META: ran handlers
redirecting (type: action) cisco.ios.ios_facts to cisco.ios.ios
Loading collection ansible.netcommon from /Users/fodorj/.ansible/collections/ansible_collections/ansible/netcommon

TASK [Gather legacy and resource facts] **************************************************************************************************************
task path: /Users/fodorj/Documents/git/ansible-network_remote_site/the_suck.yml:7
Wednesday 23 September 2020  10:33:26 -0500 (0:00:00.226)       0:00:00.226 *** 
Trying secret <ansible.parsing.vault.PromptVaultSecret object at 0x102165ca0> for vault_id=default
Trying secret <ansible.parsing.vault.PromptVaultSecret object at 0x102165ca0> for vault_id=default
Trying secret <ansible.parsing.vault.PromptVaultSecret object at 0x102165ca0> for vault_id=default
Trying secret <ansible.parsing.vault.PromptVaultSecret object at 0x102165ca0> for vault_id=default
redirecting (type: connection) ansible.builtin.network_cli to ansible.netcommon.network_cli
redirecting (type: terminal) ansible.builtin.ios to cisco.ios.ios
redirecting (type: cliconf) ansible.builtin.ios to cisco.ios.ios
<sw01-kansascity.wwt.com> attempting to start connection
<sw01-kansascity.wwt.com> using connection plugin ansible.netcommon.network_cli
Found ansible-connection at path /Users/fodorj/Downloads/exit/envs/ansible210/bin/ansible-connection
<sw01-kansascity.wwt.com> local domain socket does not exist, starting it
<sw01-kansascity.wwt.com> control socket path is /Users/fodorj/.ansible/pc/eb1106c876
<sw01-kansascity.wwt.com> redirecting (type: connection) ansible.builtin.network_cli to ansible.netcommon.network_cli
<sw01-kansascity.wwt.com> Loading collection ansible.netcommon from /Users/fodorj/.ansible/collections/ansible_collections/ansible/netcommon
<sw01-kansascity.wwt.com> redirecting (type: terminal) ansible.builtin.ios to cisco.ios.ios
<sw01-kansascity.wwt.com> Loading collection cisco.ios from /Users/fodorj/.ansible/collections/ansible_collections/cisco/ios
<sw01-kansascity.wwt.com> redirecting (type: cliconf) ansible.builtin.ios to cisco.ios.ios
<sw01-kansascity.wwt.com> Trying secret <ansible.parsing.vault.PromptVaultSecret object at 0x110d4cb20> for vault_id=default
<sw01-kansascity.wwt.com> local domain socket listeners started successfully
<sw01-kansascity.wwt.com> loaded cliconf plugin ansible_collections.cisco.ios.plugins.cliconf.ios from path /Users/fodorj/.ansible/collections/ansible_collections/cisco/ios/plugins/cliconf/ios.py for network_os ios
<sw01-kansascity.wwt.com> 
<sw01-kansascity.wwt.com> local domain socket path is /Users/fodorj/.ansible/pc/eb1106c876
Trying secret <ansible.parsing.vault.PromptVaultSecret object at 0x102165ca0> for vault_id=default
redirecting (type: action) cisco.ios.ios_facts to cisco.ios.ios
redirecting (type: action) cisco.ios.ios_facts to cisco.ios.ios
<sw01-kansascity.wwt.com> ESTABLISH LOCAL CONNECTION FOR USER: fodorj
<sw01-kansascity.wwt.com> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/fodorj/.ansible/tmp/ansible-local-16493tae1ibwt `"&& mkdir "` echo /Users/fodorj/.ansible/tmp/ansible-local-16493tae1ibwt/ansible-tmp-1600875209.442682-16498-126288275603834 `" && echo ansible-tmp-1600875209.442682-16498-126288275603834="` echo /Users/fodorj/.ansible/tmp/ansible-local-16493tae1ibwt/ansible-tmp-1600875209.442682-16498-126288275603834 `" ) && sleep 0'
<sw01-kansascity.wwt.com> Attempting python interpreter discovery
<sw01-kansascity.wwt.com> EXEC /bin/sh -c 'echo PLATFORM; uname; echo FOUND; command -v '"'"'/usr/bin/python'"'"'; command -v '"'"'python3.7'"'"'; command -v '"'"'python3.6'"'"'; command -v '"'"'python3.5'"'"'; command -v '"'"'python2.7'"'"'; command -v '"'"'python2.6'"'"'; command -v '"'"'/usr/libexec/platform-python'"'"'; command -v '"'"'/usr/bin/python3'"'"'; command -v '"'"'python'"'"'; echo ENDFOUND && sleep 0'
<sw01-kansascity.wwt.com> Python interpreter discovery fallback (unsupported platform for extended discovery: darwin)
Using module file /Users/fodorj/.ansible/collections/ansible_collections/cisco/ios/plugins/modules/ios_facts.py
<sw01-kansascity.wwt.com> PUT /Users/fodorj/.ansible/tmp/ansible-local-16493tae1ibwt/tmp1xp_0_nl TO /Users/fodorj/.ansible/tmp/ansible-local-16493tae1ibwt/ansible-tmp-1600875209.442682-16498-126288275603834/AnsiballZ_ios_facts.py
<sw01-kansascity.wwt.com> EXEC /bin/sh -c 'chmod u+x /Users/fodorj/.ansible/tmp/ansible-local-16493tae1ibwt/ansible-tmp-1600875209.442682-16498-126288275603834/ /Users/fodorj/.ansible/tmp/ansible-local-16493tae1ibwt/ansible-tmp-1600875209.442682-16498-126288275603834/AnsiballZ_ios_facts.py && sleep 0'
<sw01-kansascity.wwt.com> EXEC /bin/sh -c '/usr/bin/python /Users/fodorj/.ansible/tmp/ansible-local-16493tae1ibwt/ansible-tmp-1600875209.442682-16498-126288275603834/AnsiballZ_ios_facts.py && sleep 0'
<sw01-kansascity.wwt.com> EXEC /bin/sh -c 'rm -f -r /Users/fodorj/.ansible/tmp/ansible-local-16493tae1ibwt/ansible-tmp-1600875209.442682-16498-126288275603834/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/Users/fodorj/.ansible/tmp/ansible-local-16493tae1ibwt/ansible-tmp-1600875209.442682-16498-126288275603834/AnsiballZ_ios_facts.py", line 102, in <module>
    _ansiballz_main()
  File "/Users/fodorj/.ansible/tmp/ansible-local-16493tae1ibwt/ansible-tmp-1600875209.442682-16498-126288275603834/AnsiballZ_ios_facts.py", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/Users/fodorj/.ansible/tmp/ansible-local-16493tae1ibwt/ansible-tmp-1600875209.442682-16498-126288275603834/AnsiballZ_ios_facts.py", line 40, in invoke_module
    runpy.run_module(mod_name='ansible_collections.cisco.ios.plugins.modules.ios_facts', init_globals=None, run_name='__main__', alter_sys=True)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 188, in run_module
    fname, loader, pkg_name)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 82, in _run_module_code
    mod_name, mod_fname, mod_loader, pkg_name)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/var/folders/mf/5_fm8pgx593_82qdqb1wkrd4jrn9tj/T/ansible_cisco.ios.ios_facts_payload_uuLSgh/ansible_cisco.ios.ios_facts_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_facts.py", line 236, in <module>
  File "/var/folders/mf/5_fm8pgx593_82qdqb1wkrd4jrn9tj/T/ansible_cisco.ios.ios_facts_payload_uuLSgh/ansible_cisco.ios.ios_facts_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_facts.py", line 229, in main
  File "/var/folders/mf/5_fm8pgx593_82qdqb1wkrd4jrn9tj/T/ansible_cisco.ios.ios_facts_payload_uuLSgh/ansible_cisco.ios.ios_facts_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/facts.py", line 110, in get_facts
  File "/var/folders/mf/5_fm8pgx593_82qdqb1wkrd4jrn9tj/T/ansible_cisco.ios.ios_facts_payload_uuLSgh/ansible_cisco.ios.ios_facts_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/common/facts/facts.py", line 131, in get_network_resources_facts
  File "/var/folders/mf/5_fm8pgx593_82qdqb1wkrd4jrn9tj/T/ansible_cisco.ios.ios_facts_payload_uuLSgh/ansible_cisco.ios.ios_facts_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/acls/acls.py", line 69, in populate_facts
  File "/var/folders/mf/5_fm8pgx593_82qdqb1wkrd4jrn9tj/T/ansible_cisco.ios.ios_facts_payload_uuLSgh/ansible_cisco.ios.ios_facts_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/acls/acls.py", line 600, in render_config
  File "/var/folders/mf/5_fm8pgx593_82qdqb1wkrd4jrn9tj/T/ansible_cisco.ios.ios_facts_payload_uuLSgh/ansible_cisco.ios.ios_facts_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/acls/acls.py", line 220, in populate_port_protocol
  File "/var/folders/mf/5_fm8pgx593_82qdqb1wkrd4jrn9tj/T/ansible_cisco.ios.ios_facts_payload_uuLSgh/ansible_cisco.ios.ios_facts_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/acls/acls.py", line 140, in populate_port_protocol
RuntimeError: maximum recursion depth exceeded in cmp

**Comment by joefodor** [comment]
FYI I had to cut down on the full output since it was to long.
this output below happens 600 times
File "/var/folders/mf/5_fm8pgx593_82qdqb1wkrd4jrn9tj/T/ansible_cisco.ios.ios_facts_payload_uuLSgh/ansible_cisco.ios.ios_facts_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/acls/acls.py", line 220, in populate_port_protocol

**Comment by justjais** [comment]
@joefodor thanks for raising the issue, but for triaging it further if feasible would it possible for you to share the acl config as while parsing the config for the port protocol it's getting into recursion failure.

**Comment by synths2** [comment]
I'm experiencing the same on most of my switches but for me it doesn't seem related to ACL length. I have two switches with identical ACLs configured (only standard), one works fine and one fails with the same error as joefodor mentioned.

ACLs:
```
ip access-list standard SNMP_RO_ACCESS
 permit 10.22.2.25
 permit 10.198.135.95
 permit 10.198.135.94
 permit 10.194.207.28
 permit 10.205.216.63
 permit 10.198.135.97
 permit 10.198.135.96
 permit 10.198.135.112
 permit 10.194.124.249
 permit 10.194.124.250
 permit 10.194.124.231
 permit 10.194.124.232
 permit 10.194.124.201
 permit 10.194.124.60
 permit 10.194.124.34
 permit 10.194.35.86
 permit 10.198.163.181
 permit 10.194.124.68
ip access-list standard SNMP_RW_ACCESS
 permit 10.194.207.28
 permit 10.194.124.249
 permit 10.194.124.250
 permit 10.194.124.231
 permit 10.194.124.232
 permit 10.194.124.201
 permit 10.194.124.60
 permit 10.194.124.34
 permit 10.194.35.86
 permit 10.194.124.68
ip access-list standard VTY_ACCESS
 permit 10.27.226.169
 permit 10.27.226.120
 permit 10.27.227.19
 permit 10.206.217.241
 permit 10.206.64.0 0.0.31.255
 deny   any log
```

The working switch is a 2960xr running `15.0(2a)EX5           C2960X-UNIVERSALK9-M`. Failing switch is a 3650 running `IOS-XE 16.3.6`.
Apart from the model and OS differences I've noticed that the IOS-XE (failing) device has some additional ACLs that are built-in and do not appear in the running config (even with `show run all`). These are on all my failing devices and can be seen with `show ip access-list`, they are as follows:
```
Extended IP access list CISCO-CWA-URL-REDIRECT-ACL
    100 deny udp any any eq domain
    101 deny tcp any any eq domain
    102 deny udp any eq bootps any
    103 deny udp any any eq bootpc
    104 deny udp any eq bootpc any
    105 permit tcp any any eq www
Extended IP access list preauth_ipv4_acl (per-user)
    10 permit udp any any eq domain
    20 permit tcp any any eq domain
    30 permit udp any eq bootps any
    40 permit udp any any eq bootpc
    50 permit udp any eq bootpc any
    60 deny ip any any
```

I'm not sure if those are something to do with the problem or if it's more related to model/version. Thanks.

**Comment by synths2** [comment]
I'm experiencing the same on most of my switches but for me it doesn't seem related to ACL length. I have two switches with identical ACLs configured (only standard), one works fine and one fails with the same error as joefodor mentioned.

ACLs:
```
ip access-list standard SNMP_RO_ACCESS
 permit 10.22.2.25
 permit 10.198.135.95
 permit 10.198.135.94
 permit 10.194.207.28
 permit 10.205.216.63
 permit 10.198.135.97
 permit 10.198.135.96
 permit 10.198.135.112
 permit 10.194.124.249
 permit 10.194.124.250
 permit 10.194.124.231
 permit 10.194.124.232
 permit 10.194.124.201
 permit 10.194.124.60
 permit 10.194.124.34
 permit 10.194.35.86
 permit 10.198.163.181
 permit 10.194.124.68
ip access-list standard SNMP_RW_ACCESS
 permit 10.194.207.28
 permit 10.194.124.249
 permit 10.194.124.250
 permit 10.194.124.231
 permit 10.194.124.232
 permit 10.194.124.201
 permit 10.194.124.60
 permit 10.194.124.34
 permit 10.194.35.86
 permit 10.194.124.68
ip access-list standard VTY_ACCESS
 permit 10.27.226.169
 permit 10.27.226.120
 permit 10.27.227.19
 permit 10.206.217.241
 permit 10.206.64.0 0.0.31.255
 deny   any log
```

The working switch is a 2960xr running `15.0(2a)EX5           C2960X-UNIVERSALK9-M`. Failing switch is a 3650 running `IOS-XE 16.3.6`.
Apart from the model and OS differences I've noticed that the IOS-XE (failing) device has some additional ACLs that are built-in and do not appear in the running config (even with `show run all`). These are on all my failing devices and can be seen with `show ip access-list`, they are as follows:
```
Extended IP access list CISCO-CWA-URL-REDIRECT-ACL
    100 deny udp any any eq domain
    101 deny tcp any any eq domain
    102 deny udp any eq bootps any
    103 deny udp any any eq bootpc
    104 deny udp any eq bootpc any
    105 permit tcp any any eq www
Extended IP access list preauth_ipv4_acl (per-user)
    10 permit udp any any eq domain
    20 permit tcp any any eq domain
    30 permit udp any eq bootps any
    40 permit udp any any eq bootpc
    50 permit udp any eq bootpc any
    60 deny ip any any
```

I'm not sure if those are something to do with the problem or if it's more related to model/version. Thanks.

**Comment by joefodor** [comment]
@justjais thank you for responding! because of security concerns I have changed ips to x.x.x.x I hope that is ok.

ip access-list extended 185
 permit icmp any any
 permit igmp any any
 permit pim any any
 permit udp any any eq bootps
 permit udp any any eq bootpc
 permit udp any host x.x.x.x eq 1985
 permit eigrp any any
 permit udp any host x.x.x.x eq domain
 permit tcp any host x.x.x.x eq domain
 permit udp any host x.x.x.x eq domain
 permit tcp any host x.x.x.x eq domain
 permit udp any host x.x.x.x eq domain
 permit tcp any host x.x.x.x eq domain
 permit udp any host x.x.x.x eq domain
 permit tcp any host x.x.x.x eq domain
 permit udp any host x.x.x.x eq domain
 permit tcp any host x.x.x.x eq domain
 permit udp any host x.x.x.x eq domain
 permit tcp any host x.x.x.x eq domain
 permit udp any host x.x.x.x eq netbios-ns
 permit udp any host x.x.x.x eq netbios-ns
 permit udp any host x.x.x.x eq netbios-ns
 permit udp any host x.x.x.x eq netbios-ns
 permit udp any host x.x.x.x eq netbios-ns
 permit udp any host x.x.x.x eq netbios-ns
 permit udp any host x.x.x.x eq ntp
 permit udp any host x.x.x.x eq ntp
 permit tcp any any eq 2000
 permit udp any any range x.x.x.x
 permit udp any any eq 5445
 permit udp any any gt 16384
 permit udp any host x.x.x.x eq 1820
 permit tcp any any eq 5800
 permit tcp any any eq 5900
 permit tcp any eq 7775 any
 permit tcp any host x.x.x.x eq 992
 permit tcp any host x.x.x.x eq telnet
 permit tcp any host x.x.x.x eq 992
 permit tcp any host x.x.x.x eq telnet
 permit tcp any eq telnet any
 permit tcp any eq www any log
 permit tcp any eq lpd any log
 permit tcp any eq 9100 any log
 permit udp any eq snmp any
 permit tcp any eq 443 any log
 permit tcp any host x.x.x.x eq 992
 permit tcp any host x.x.x.x eq telnet
 permit tcp any host x.x.x.x eq 992
 permit tcp any host x.x.x.x eq telnet
 permit tcp any host x.x.x.x eq 992
 permit tcp any host x.x.x.x eq telnet
 permit tcp any host x.x.x.x eq ftp
 permit ip any host x.x.x.x
 permit tcp any x.x.x.x x.x.x.x eq www
 permit tcp any x.x.x.x x.x.x.x eq 443
 permit tcp any x.x.x.x x.x.x.x eq www
 permit tcp any x.x.x.x x.x.x.x eq 443
 permit tcp any x.x.x.x x.x.x.x eq www
 permit tcp any x.x.x.x x.x.x.x eq 443
 permit tcp any x.x.x.x x.x.x.x eq www
 permit tcp any x.x.x.x x.x.x.x eq 443
 permit tcp any x.x.x.x x.x.x.x eq www
 permit tcp any x.x.x.x x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any x.x.x.x x.x.x.x eq www
 permit tcp any x.x.x.x x.x.x.x eq 443
 permit tcp any x.x.x.x x.x.x.x eq www
 permit tcp any x.x.x.x x.x.x.x eq 443
 permit tcp any x.x.x.x x.x.x.x eq www
 permit tcp any x.x.x.x x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit ip any host x.x.x.x
 permit ip any host x.x.x.x
 permit ip any host x.x.x.x
 permit ip any host x.x.x.x
 permit ip any host x.x.x.x
 permit ip any host x.x.x.x
 permit ip any host x.x.x.x
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit udp any host x.x.x.x eq 443
 permit udp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 8443
 permit udp any host x.x.x.x eq 8443
 permit tcp any host x.x.x.x eq 8443
 permit udp any host x.x.x.x eq 8443
 permit tcp any host x.x.x.x eq 18287
 permit tcp any host x.x.x.x eq whois
 permit tcp any host x.x.x.x eq whois
 permit tcp any host x.x.x.x eq whois
 permit tcp any host x.x.x.x eq whois
 permit tcp any host x.x.x.x eq whois
 permit tcp any host x.x.x.x eq whois
 permit tcp any host x.x.x.x eq whois
 permit tcp any host x.x.x.x eq whois
 permit tcp any host x.x.x.x eq whois
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq whois
 permit tcp any host x.x.x.x eq www
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 5228
 permit tcp any host x.x.x.x eq 5229
 permit tcp any host x.x.x.x eq 5230
 permit tcp any host x.x.x.x eq 443
 permit tcp any x.x.x.x x.x.x.x eq www
 permit tcp any x.x.x.x x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 11200
 permit tcp any host x.x.x.x eq 11200
 permit tcp any host x.x.x.x eq 11200
 permit tcp any host x.x.x.x eq 11200
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 11200
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 443
 permit tcp any host x.x.x.x eq 11200
 permit tcp any host x.x.x.x eq 11200
 permit tcp any host x.x.x.x eq 55498
 deny   ip any any log-input

**Comment by justjais** [comment]
@joefodor Thanks so much for the required info, I'll triage and update the issue asap.

**Comment by joefodor** [comment]
@justjais I wanted to share with you some findings on this issue.

I took that really long ACL and started pairing down until I got it to run. I found that on like 60 it fails. When I remove line 60 it works. It doesn't seem to be an issue with the length but something along the lines of syntax. I can put in a permit ip any any on line 60 and it will pass. 


fails:
Extended IP access list 185
    20 permit igmp any any
    30 permit pim any any
    40 permit udp any any eq bootps
    50 permit udp any any eq bootpc
    60 permit udp any host 224.0.0.2 eq 1985
    70 permit eigrp any any

pass:
Extended IP access list 185
    20 permit igmp any any
    30 permit pim any any
    40 permit udp any any eq bootps
    50 permit udp any any eq bootpc
    70 permit eigrp any any

I then realized the only difference in that line is that its using 'host'
